frontend/src/app/projects/models/model-source/t4c/manage-t4c-model/manage-t4c-model.component.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, switchMap, tap } from 'rxjs';
+import { ToastService } from 'src/app/helpers/toast/toast.service';
+import { notNullValidator } from 'src/app/helpers/validators/validators';
+import {
+  SubmitT4CModel,
+  T4CModel,
+  T4CModelService,
+} from 'src/app/projects/models/model-source/t4c/service/t4c-model.service';
+import { ModelService } from 'src/app/services/model/model.service';
+import { ProjectService } from 'src/app/services/project/project.service';
+import {
+  T4CInstance,
+  T4CInstanceService,
+} from 'src/app/services/settings/t4c-instance.service';
+import {
+  T4CRepoService,
+  T4CRepository,
+} from 'src/app/settings/modelsources/t4c-settings/service/t4c-repos/t4c-repo.service';
+
+@Component({
+  selector: 'app-manage-t4c-model',
+  templateUrl: './manage-t4c-model.component.html',
+  styleUrls: ['./manage-t4c-model.component.css'],
+})
+export class ManageT4CModelComponent implements OnInit, OnDestroy {
+  @Input() asStepper?: boolean;
+  @Output() create = new EventEmitter<boolean>();
+
+  editing = false;
+  loading = false;
+
+  t4cInstances: T4CInstance[] | undefined = undefined;
+  get selectedT4CInstance(): T4CInstance | null | undefined {
```

Comment:
Why do you need the undefined case here? When I needed to handle lists of resources in some components, I just assigned the empty list as the default value, since this doesn't really break anything and you have reduced complexity by not having to consider the extra case (e.g. you could remove the `null` return type for `get selectedT4CInstance`).
The same also applies for the `t4cRepositories`

Reply:
Having an empty list as default value is bad style imho. An empty list means there is no T4C instance - which is different to the undefined case, which means "not loaded yet" / "not defined yet". In some parts I use this to indicate the loading animation -> If it is undefined, it shows the loading animation, otherwise not.

But I just noticed that `selectedT4CInstance` is not used. Removed it.